data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/coordination/partition/StreamPartition.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.dynamodb.coordination.partition;
+
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.SourcePartition;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.state.StreamProgressState;
+
+import java.util.Optional;
+
+public class StreamPartition extends SourcePartition<StreamProgressState> {
+
+    public static final String PARTITION_TYPE = "STREAM";
+
+    private final String streamArn;
+
+    private final String shardId;
+
+    private final Optional<StreamProgressState> state;
+
+    public StreamPartition(String streamArn, String shardId, Optional<StreamProgressState> state) {
```

Comment:
Again, do not pass in an `Optional`. This can be a nullable `StreamProgressState`.

You can't be sure that the `Optional` itself is not null, so you would need a null-check to be correct.